custom_components/ha_washdata/suggestion_engine.py

```
@@ -447,9 +1221,283 @@ def run_batch_simulation(self, cycles: list[dict[str, Any]]) -> dict[str, Any]:
+                in_low = True
+                low_start_s = t
+            elif in_low and p >= active_thr:
+                dur = points[i - 1][0] - low_start_s
+                if dur >= 30.0:  # ignore motor micro-dips
+                    score: float | None = None
+                    try:
+                        feat = end_feat_fn(points[:i], expectation)  # tail is the low run
+                        if feat is not None:
+                            score = float(end_score_fn(feat))
+                    except Exception:  # pylint: disable=broad-exception-caught
+                        pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)